--- 4.2r1-hrepack-p4/mfhdf/libsrc/file.c.maxavailfiles	2007-04-20 12:21:12.000000000 -0600
+++ 4.2r1-hrepack-p4/mfhdf/libsrc/file.c	2007-04-20 12:21:50.000000000 -0600
@@ -31,7 +31,7 @@
 
 /* Maximum number of files can be opened at one time; subtract 3 from
    the system allowed to account for stdin, stdout, and stderr */
-#define MAX_AVAIL_OPENFILES          (MAX_SYS_OPENFILES - 3)
+#define MAX_AVAIL_OPENFILES          (MAX_SYS_OPENFILES - 10)
 
 static int _ncdf = 0 ; /*  high water mark on open cdf's */
 static NC **_cdfs;
