--- HDF4.2r4/mfhdf/libsrc/tncunlim.c.buffer	2009-01-10 19:34:26.000000000 -0700
+++ HDF4.2r4/mfhdf/libsrc/tncunlim.c	2009-02-25 11:08:46.000000000 -0700
@@ -646,7 +646,7 @@
     long  dimsize = 0,		/* unlimited dimension size */
 	  dimsize1 = 0, dimsize2 = 0; /* other dimension sizes */
     short outdata[DIM0][DIM1][DIM2];	/* data read back */
-    char  varname[10];		/* variable name */
+    char  varname[12];		/* variable name */
     nc_type rh_type;		/* variable type */
     int   rh_ndims;		/* number of dims */
     int   rh_dims[H4_MAX_VAR_DIMS];	/* variable shape */
